### ISSUE TITLE
Unable to remove livenessProbe using server-side apply

### ISSUE THREAD

**Issue by cbandy** [kubernetes/kubernetes#99037]
#### What happened:

I removed a `livenessProbe` from my SSA intent, but could not apply it due to validation errors.

#### What you expected to happen:

I expected the probe to be removed because there were no more managers of it.

https://kubernetes.io/docs/reference/using-api/server-side-apply/
> If you remove a field from a configuration and apply the configuration, server side apply checks if there are any other field managers that also own the field. If the field is not owned by any other field managers, it is either deleted from the live object or reset to its default value, if it has one.

#### How to reproduce it (as minimally and precisely as possible):

Create an object that has a `livenessProbe` in its PodTemplateSpec; a ReplicaSet here:

```yaml
#$ cat rs-probe.yaml
apiVersion: apps/v1
kind: ReplicaSet
metadata:
  name: pods
spec:
  replicas: 1
  selector:
    matchLabels: { name: pods }
  template:
    metadata:
      labels: { name: pods }
    spec:
      containers:
        - name: busybox
          image: docker.io/library/busybox
          command: ['tail', '-f', '/dev/null']
          livenessProbe: { exec: { command: ['/bin/true'] } }
```

Apply it and see that the probe is managed and present:

```
$ kubectl apply --server-side -f rs-probe.yaml
replicaset.apps/pods serverside-applied

$ kubectl get rs/pods -o yaml | grep -A1 Probe
                f:livenessProbe:
                  f:exec:
--
        livenessProbe:
          exec:
```

Comment out or remove the probe:

```
$ tail -n2 rs-probe.yaml
          command: ['tail', '-f', '/dev/null']
          #livenessProbe: { exec: { command: ['/bin/true'] } }
```

Attempt to apply it:

```
$ kubectl apply --server-side -f rs-probe.yaml
The ReplicaSet "pods" is invalid: spec.template.spec.containers[0].livenessProbe: Required value: must specify a handler type
```

#### Environment:
- Kubernetes version (use `kubectl version`):

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2", GitCommit:"faecb196815e248d3ecfb03c680a4507229c2a56", GitTreeState:"clean", BuildDate:"2021-01-13T13:28:09Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2", GitCommit:"faecb196815e248d3ecfb03c680a4507229c2a56", GitTreeState:"clean", BuildDate:"2021-01-13T13:20:00Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
```

#### Anything else we need to know?:

With Kubernetes v1.18.15, the behavior is slightly different. The second SSA (sans probe) succeeds and the field stops being managed _but_ it remains stored in the spec:

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.15", GitCommit:"73dd5c840662bb066a146d0871216333181f4b64", GitTreeState:"clean", BuildDate:"2021-01-13T13:22:41Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.15", GitCommit:"73dd5c840662bb066a146d0871216333181f4b64", GitTreeState:"clean", BuildDate:"2021-01-13T13:14:05Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}

$ tail -n2 rs-probe.yaml
          command: ['tail', '-f', '/dev/null']
          #livenessProbe: { exec: { command: ['/bin/true'] } }

$ kubectl apply --server-side -f rs-probe.yaml
replicaset.apps/pods serverside-applied

$ kubectl get rs/pods -o yaml | grep -A1 Probe
        livenessProbe:
          exec:
```

**Comment by apelisse** [comment]
I've literally just fixed that bug! #99014

**Comment by apelisse** [comment]
I'll try and test that exact use-case to see if my fix works well ;-)

**Comment by apelisse** [comment]
OK Just verified, should work fine!

**Comment by apelisse** [comment]
Fixed by #99014

**Comment by nikhita** [comment]
Closing since it's fixed by https://github.com/kubernetes/kubernetes/pull/99014. Please reopen if needed! :)
